test/functional/test-ad-cid.js

```diff
@@ -25,224 +25,136 @@ import {setCookie} from '../../src/cookies';
-          height: 250,
-          type: '_ping_',
-          src: 'testsrc',
-          'data-consent-notification-id': 'uid',
-        }, 'https://schema.org', function(ad) {
-          const win = ad.ownerDocument.defaultView;
-          const cidService = installCidService(win);
-          const uidService = installUserNotificationManager(win);
-          sandbox.stub(uidService, 'get', id => {
-            expect(id).to.equal('uid');
-            return Promise.resolve('consent');
-          });
-          sandbox.stub(cidService, 'get', (scope, consent) => {
-            expect(scope).to.equal(cidScope);
-            return consent.then(val => {
-              return val + '-cid';
-            });
+        sandbox.stub(cidService, 'get', (scope, consent) => {
+          expect(scope).to.equal(cidScope);
+          return consent.then(val => {
+            return val + '-cid';
           });
-          return ad;
-        }).then(ad => {
-          const src = ad.firstChild.getAttribute('src');
-          expect(src).to.contain('"clientId":"consent"');
         });
-      });
-
-      it('provide null if notification and cid is not provided', () => {
-        config.clientIdScope = null;
-        let uidSpy = null;
-        return getAd({
-          width: 300,
-          height: 250,
-          type: '_ping_',
-          src: 'testsrc',
-        }, 'https://schema.org', function(ad) {
-          const win = ad.ownerDocument.defaultView;
-          const cidService = installCidService(win);
-          const uidService = installUserNotificationManager(win);
-          uidSpy = sandbox.spy(uidService, 'get');
-          sandbox.stub(cidService, 'get', (scope, consent) => {
-            expect(scope).to.equal(cidScope);
-            return consent.then(val => {
-              return val + '-cid';
-            });
-          });
-          return ad;
-        }).then(ad => {
-          expect(uidSpy.callCount).to.equal(0);
-          const src = ad.firstChild.getAttribute('src');
-          expect(src).to.contain('"clientId":null');
+        const uidSpy = sandbox.spy(uidService, 'get');
+        return getAdCid(adElement).then(cid => {
+          expect(uidSpy).to.not.be.called;
+          expect(cid).to.be.undefined;
         });
       });
 
-      it('provides null if cid service not available', () => {
-        config.clientIdScope = cidScope;
-        return getAd({
-          width: 300,
-          height: 250,
-          type: '_ping_',
-          src: 'testsrc',
-        }, 'https://schema.org', function(ad) {
-          setCookie(window, cidScope, 'XXX', Date.now() + 5000);
-          return ad;
-        }).then(ad => {
-          const src = ad.firstChild.getAttribute('src');
-          expect(src).to.contain('"clientId":null');
-        });
+  it('should return null if cid service not available', () => {
+    config.clientIdScope = cidScope;
+    return createIframePromise(true /* runtimeOff */).then(iframe => {
```

Comment:
I am installing cid service in describe. But I need to mock the case that runtime never install cid service. createIframePromise(true /*runtimeOff*/) is the perfect helper function here. Otherwise I can't find a way to uninstall cid service and test.